### PR TITLE
Update logits_process.py docstrings to clarify penalty and reward cases

### PR DESCRIPTION
**Pull request by larekrow**
This PR fixes point 3 of #25970 by clarifying the penalty and reward cases for `RepetitionPenaltyLogitsProcessor` and `EncoderRepetitionPenaltyLogitsProcessor` within the docstrings.

`EncoderRepetitionPenaltyLogitsProcessor` may require further iterations as the class name does not accurately reflect what the class is doing.

@gante

### CODE REVIEW

File: src/transformers/generation/logits_process.py [hunk 1]
```diff
@@ -276,9 +276,15 @@ class RepetitionPenaltyLogitsProcessor(LogitsProcessor):
     selected. The formula can be seen in the original [paper](https://arxiv.org/pdf/1909.05858.pdf). According to the
     paper a penalty of around 1.2 yields a good balance between truthful generation and lack of repetition.
 
+    While the intended usage within the paper is to penalize and thus reduce repetition in the generated sequence, this
+    technique can also be used to reward and thus encourage repetition in a similar manner. To penalize and reduce
```
Comment (gante) [inline src/transformers/generation/logits_process.py:280]:
```suggestion
    This technique can also be used to reward and thus encourage repetition in a similar manner. To penalize and reduce 
```
1. Let's trim the intended usage part, most users are only concerned about practical uses :) Short and clear docstring = better!
2. You may need to run `make fixup` after applying the suggested change :)

File: src/transformers/generation/logits_process.py [hunk 2]
```diff
@@ -322,11 +328,18 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
 
 class EncoderRepetitionPenaltyLogitsProcessor(LogitsProcessor):
     r"""
-    [`LogitsProcessor`] enforcing an exponential penalty on tokens that are not in the original input.
+    [`LogitsProcessor`] enforcing an exponential penalty on tokens that are not in the original input. This technique
+    avoids hallucination by boosting the probabilities of tokens found within the original input.
+
+    While the intended usage is to penalize and thus reduce hallucination, this technique can also be used to reward and
+    thus encourage creativity in a similar manner. To penalize and reduce repetition, use `hallucination_penalty` values
```
Comment (gante) [inline src/transformers/generation/logits_process.py:335]:
```suggestion
    This technique can also be used to reward and thus encourage creativity in a similar manner. To penalize and reduce repetition, use `hallucination_penalty` values
```
(same comment as above)

File: src/transformers/generation/logits_process.py [hunk 3]
```diff
@@ -322,11 +328,18 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
...
+    above 1.0, where a higher value penalizes more strongly. To reward and encourage creativity, use 
+    `hallucination_penalty` values between 0.0 and 1.0, where a lower value rewards more strongly.
 
     Args:
         hallucination_penalty (`float`):
```
Comment (gante) [inline src/transformers/generation/logits_process.py:340]: Proof of my poor past review: this arg should match the one in `__init__`. Would you be so kind as to update `hallucination_penalty` to `penalty` here?

(we can't do it the other way around because of backwards compatibility :) )

**Review by gante — COMMENTED** [review]
Thank you for the clarifications! A few suggestions :)

**Comment by larekrow** [comment]
Hey @gante @ArthurZucker! This is ready for review 😃
